zebra/kernel_socket.c

```diff
@@ -1390,13 +1395,22 @@ static void routing_socket(struct zebra_ns *zns)
 {
 	frr_elevate_privs(&zserv_privs) {
 		routing_sock = ns_socket(AF_ROUTE, SOCK_RAW, 0, zns->ns_id);
+
+		dplane_routing_sock =
+			ns_socket(AF_ROUTE, SOCK_RAW, 0, zns->ns_id);
 	}
 
 	if (routing_sock < 0) {
 		flog_err_sys(EC_LIB_SOCKET, "Can't init kernel routing socket");
 		return;
 	}
 
+	if (dplane_routing_sock < 0) {
+		flog_err_sys(EC_LIB_SOCKET,
+			     "Can't init kernel dataplane routing socket");
+		return;
+	}
+
 	/* XXX: Socket should be NONBLOCK, however as we currently
```

Comment:
caution. dplane_routing_sock should not be kept global.
in the case of routing_sock, the variable is copied as parameter into thread.
but for dplane_routing_sock, this does not operate like that.

so I wonder , what happens if I have a vrf backend with network namespaces.
Will we not lose some information ( because the socket integer will be overwritten by the latter socket) ?

Reply:
I'm not sure I understand what you mean be "global" there - each zebra namespace object gets its own copy of the three netlink sockets?

Reply:
actually, the path used here is not the netlink path socket. this is the routing socket path which is not know what a namespace is.